oms/azuredeploy.json

```diff
@@ -0,0 +1,1210 @@
+        "url": "https://devopsgallerystorage.blob.core.windows.net/packages/azurerm.recoveryservices.1.1.3.nupkg"
+      },
+      "azureRmBackup": {
+        "name": "AzureRm.Backup",
+        "url": "https://devopsgallerystorage.blob.core.windows.net/packages/azurerm.backup.1.0.9.nupkg"
+      },
+      "azureRmCompute": {
+        "name": "AzureRm.Compute",
+        "url": "https://devopsgallerystorage.blob.core.windows.net/packages/azurerm.compute.1.3.3.nupkg"
+      },
+      "azureRmAutomation": {
+        "name": "AzureRm.Automation",
+        "url": "https://devopsgallerystorage.blob.core.windows.net/packages/azurerm.automation.1.0.11.nupkg"
+      }
+    },
+    "runbooks": {
+      "omsABUscript": {
+        "name": "AzureIaasBackup",
+        "version": "1.0.0.0",
+        "description": "Runbook to automatically backup IaaS VMs within the ABU region",
+        "type": "PowerShell",
+        "Id": ""
+      }
+    },
+    "_assetLocation": "https://raw.githubusercontent.com/krnese/AzureDeploy/master/OMS/MSOMS/AzureIaaSBackup",
+    "scriptUri": "[concat(variables('_assetLocation'),'/scripts/Automated-IaaS-Backup.ps1')]",
+    "ingestInterval": "1",
+    "ingestFrequency": "hour",
+    "azureSubscriptionId": "AzureSubscriptionId",
+    "omsWorkspaceId": "OMSWorkspaceId",
+    "omsWorkspaceKey": "OMSWorkspacekey",
+    "omsWorkspaceName": "OMSWorkspaceName",
+    "omsResourceGroupName": "OMSResourceGroupName",
+    "omsRecoveryVault": "OMSRecoveryVault",
+    "recoveryStorageAccountName": "[toLower(concat('asr', uniqueString(resourceGroup().id)))]",
+    "recoveryStorageType": "Standard_LRS",
+    "securitySolution": "[concat('Security', '(', parameters('omsWorkspaceName'), ')')]",
+    "securitySolutionName": "Security",
+    "agentSolution": "[concat('AgentHealthAssessment', '(', parameters('omsWorkspaceName'), ')')]",
+    "agentSolutionName": "AgentHealthAssessment",
+    "changeTrackingSolution": "[concat('ChangeTracking', '(', parameters('omsWorkspaceName'), ')')]",
+    "changeTrackingSolutionName": "ChangeTracking",
+    "updatesSolution": "[concat('Updates', '(', parameters('omsWorkspaceName'), ')')]",
+    "updatesSolutionName": "Updates"
+  },
+  "resources": [
+    {
+      "apiVersion": "2016-06-01",
+      "location": "[parameters('omsRecoveryVaultRegion')]",
+      "name": "[parameters('omsRecoveryVaultName')]",
+      "type": "Microsoft.RecoveryServices/vaults",
+      "comments": "ASR Recovery Vault for OMS",
+      "sku": {
+        "name": "RS0",
+        "tier": "Standard"
+      },
+      "properties": { },
+      "resources": [
+        {
+          "apiVersion": "2016-06-01",
+          "name": "HyperVPrimarySite",
+          "type": "replicationFabrics",
+          "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.RecoveryServices/vaults/', parameters('omsRecoveryVaultName'), '/replicationFabrics/', 'HyperVPrimarySite')]",
```

Comment:
is there a reason not to use the [resourceId function](https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#resourceid)? Would make this simpler